controllers/envoy/envoyconfig_controller.go

```diff
@@ -60,34 +58,21 @@ func (r *EnvoyConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, err
 	}
 
-	// Add finalizer for this CR
-	if !contains(ec.GetFinalizers(), envoyv1alpha1.EnvoyConfigFinalizer) {
-		r.Log.Info("Adding Finalizer for the EnvoyConfig")
-		if err := r.addFinalizer(ctx, ec); err != nil {
-			r.Log.Error(err, "Failed adding finalizer for envoyconfig")
-			return ctrl.Result{}, err
-		}
+	// Set defaults.
+	// TODO: remove this when wwe migrate to CRD v1 api as we will
+	// be able to set defauls directly in the CRD definition.
+	if err := r.reconcileSpecDefaults(ctx, ec); err != nil {
+		return ctrl.Result{}, err
 	}
```

Comment:
This is mainly done to fill the `spec.envoyAPI` field if it is left unset by the user so list operations with kubectl show the correct information. Won't be necessary when upgrading the CRD api as it will directly accept specifying default values.